lambda-functions/cleanup-unused-images/main.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	log "github.com/sirupsen/logrus"
+)
+
+func main() {
+	log.SetLevel(log.DebugLevel)
+
+	lambda.Start(handler)
+}
+
+func handler() {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(os.Getenv("AWS_REGION"))},
+	)
+	if err != nil {
+		log.WithError(err).Error("AWS Session failed.")
+		return
+	}
+	svc := ec2.New(sess)
+	uniqueUsedImages, err := getUniqueUsedImages(svc)
+	if err != nil {
+		log.WithError(err).Error("Failed to get Unique Used AMIs.")
+		return
+	}
+	err = deleteAMIs(svc, uniqueUsedImages)
+
+	if err != nil {
+		log.WithError(err).Error("Failed to delete AMIs.")
+		return
+	}
+}
+
+func deleteAMIs(svc *ec2.EC2, uniqueUsedImages []string) error {
+	imagesInput := &ec2.DescribeImagesInput{
+		Owners: []*string{
+			aws.String(os.Getenv("OWNER_ID")),
+		},
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("tag:Name"),
+				Values: []*string{aws.String("mattermost-cloud-*")},
+			},
+		},
+	}
+	snapshots, err := getAllSnapshots(os.Getenv("OWNER_ID"), svc)
+	if err != nil {
+		return errors.Wrap(err, "Failed to get Snapshots.")
+	}
+	allImages, err := svc.DescribeImages(imagesInput)
+	if err != nil {
+		return errors.Wrap(err, "Failed to describe images.")
+	}
+	oldImages, err := filterImagesByDateRange(allImages.Images, 730)
+	if err != nil {
+		return errors.Wrap(err, "Failed to filter images by date range.")
+	}
+	dryRun := true
+	for _, i := range oldImages {
+		imageForCleanup := contains(uniqueUsedImages, *i.ImageId)
+		if imageForCleanup != "" {
+			fmt.Println(*i.ImageId + ": De-registering AMI named \"" + *i.Name + "\"...")
```

Comment:
nit: there are a few remaining `fmt.Println` that could be replaced with `log.Info`, but not a big deal.

Reply:
@gabrieljackson fixed ;)